solr/core/src/java/org/apache/solr/handler/ContentStreamHandlerBase.java

```diff
@@ -63,6 +70,17 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
           SolrException.ErrorCode.SERVER_ERROR,
           "Updates are temporarily paused for core: " + req.getCore().getName());
     }
+    List<CircuitBreaker> trippedCircuitBreakers =
+        req.getCore().getCircuitBreakerRegistry().checkTripped(RequestType.UPDATE);
+    if (trippedCircuitBreakers != null) {
+      String errorMessage = CircuitBreakerRegistry.toErrorMessage(trippedCircuitBreakers);
+      rsp.add(STATUS, FAILURE);
+      rsp.setException(
+          new SolrException(
+              SolrException.ErrorCode.SERVICE_UNAVAILABLE,
+              "Circuit Breakers tripped " + errorMessage));
+      return;
```

Comment:
I think a `solrCoreState.deregisterInFlightUpdate();` call is needed before returning, or for the circuit breaker to be checked first before registering the update as in-flight.

Reply:
Placed it on top before registering.